File: Java/SimpleReranker.java
package mu.cse.rag.clean;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class SimpleReranker implements Reranker {
	@Override
    public List<Hit> rerank(List<String> terms, List<Hit> hits) {
        List<Hit> mutableHits = new ArrayList<>(hits);

        for (Hit hit : mutableHits) {
            String title = hit.chunk.title.toLowerCase(Locale.forLanguageTag("tr-TR"));
            String text = hit.chunk.text.toLowerCase(Locale.forLanguageTag("tr-TR"));
            int score = 0; 
  
            for (String term : terms) { //Add points for each term found
                if (text.contains(term)) {
                    score += 1;
                }
                if (title.contains(term)) { //Add more points for title matches
                    score += 3;
                }
            }

            String query = String.join(" ", terms);
            if (text.contains(query)) {
                score += 5; 
            }
            hit.score = score;
        }

        mutableHits.sort((h1, h2) -> Integer.compare(h2.score, h1.score));
        return mutableHits;
    }
}

File: JavaToPython/SimpleReranker.py
from typing import List


class SimpleReranker:
    def rerank(self, terms: List[str], hits: List['Hit']) -> List['Hit']:
        mutable = list(hits)
        lower_terms = [t.lower() for t in terms]
        for hit in mutable:
            title = (hit.chunk.title or '').lower()
            text = (hit.chunk.text or '').lower()
            score = 0
            for term in lower_terms:
                if term and term in text:
                    score += 1
                if term and term in title:
                    score += 3
            query = ' '.join(lower_terms)
            if query and query in text:
                score += 5
            hit.score = score
        mutable.sort(key=lambda h: h.score, reverse=True)
        return mutable

